x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/machine_learning_alert_suppression.ts

```diff
@@ -0,0 +1,1082 @@
+            ...baseAnomaly,
+            timestamp,
+            'host.name': ['relevant'],
+          };
+          const anomalyWithoutSuppressionField = {
+            ...baseAnomaly,
+            timestamp,
+          };
+          await indexListOfDocuments([anomaly, anomaly, anomalyWithoutSuppressionField]);
+
+          const rule = {
+            ...ruleProps,
+            alert_suppression: {
+              ...ruleProps.alert_suppression,
+              group_by: ['host.name'],
+            },
+          };
+          const { previewId } = await previewRule({
+            supertest,
+            rule,
+            timeframeEnd: new Date(timestamp),
+            invocationCount: 1,
+          });
+          const previewAlerts = await getPreviewAlerts({
+            es,
+            previewId,
+            sort: [ALERT_SUPPRESSION_DOCS_COUNT],
+          });
+
+          expect(previewAlerts.length).toEqual(2);
+          expect(previewAlerts[0]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [
+                {
+                  field: 'host.name',
+                  value: null,
+                },
+              ],
+              [TIMESTAMP]: timestamp,
+              [ALERT_START]: timestamp,
+              [ALERT_ORIGINAL_TIME]: timestamp,
+              [ALERT_SUPPRESSION_START]: timestamp,
+              [ALERT_SUPPRESSION_END]: timestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 0,
+            })
+          );
+
+          expect(previewAlerts[1]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [
+                {
+                  field: 'host.name',
+                  value: ['relevant'],
+                },
+              ],
+              [TIMESTAMP]: timestamp,
+              [ALERT_START]: timestamp,
+              [ALERT_ORIGINAL_TIME]: timestamp,
+              [ALERT_SUPPRESSION_START]: timestamp,
+              [ALERT_SUPPRESSION_END]: timestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 1, // the anomaly without `host.name` is not represented here
+            })
+          );
+        });
+
+        it('does not suppress alerts with missing fields, if not configured to do so', async () => {
+          const rule = {
+            ...ruleProps,
+            alert_suppression: {
+              ...ruleProps.alert_suppression,
+              group_by: ['host.name'],
+              missing_fields_strategy: 'doNotSuppress' as const,
+            },
+          };
+          const timestamp = new Date().toISOString();
+          const anomaly = {
+            ...baseAnomaly,
+            timestamp,
+            'host.name': ['relevant'],
+          };
+          const anomalyWithoutSuppressionField = {
+            ...baseAnomaly,
+            timestamp,
+            'user.name': ['irrelevant'],
+          };
+          await indexListOfDocuments([anomaly, anomaly, anomalyWithoutSuppressionField]);
```

Comment:
would be better to have 2 anomalyWithoutSuppressionField here, so we would clearly see there 2 generated alerts per each anomaly

Reply:
ce395da6ae7